redis/src/main/scala/zio/redis/RespValue.scala

```diff
@@ -17,12 +17,78 @@
 package zio.redis
 
 import zio._
+import zio.redis.RespValue.BulkString
 import zio.redis.codecs.CRC16
 import zio.redis.options.Cluster.Slot
+import zio.schema.Schema
+import zio.schema.codec.BinaryCodec
 import zio.stream._
 
 import java.nio.charset.StandardCharsets
 
+final case class RespCommand(respArgs: Chunk[RespArgument]) {
+  def ++(that: RespCommand): RespCommand = RespCommand(this.respArgs ++ that.respArgs)
+
+  def transformArgs(f: RespArgument => RespArgument): RespCommand = RespCommand(respArgs.map(f(_)))
+}
+
+object RespCommand {
+
+  def empty: RespCommand = new RespCommand(Chunk.empty)
+
+  def apply(respArgs: Chunk[RespArgument]): RespCommand = new RespCommand(respArgs)
+
+  def apply(respArgs: RespArgument*): RespCommand = new RespCommand(Chunk.fromIterable(respArgs))
+
+  def apply(respArg: RespArgument): RespCommand = new RespCommand(Chunk.single(respArg))
+}
+
+sealed trait RespArgument {
+  def respValue: RespValue.BulkString
+}
```

Comment:
If you would like to proceed with this, please:
- submit it in a separate pull request as it is not related to slots and keys
- either split it to multiple files or respect scala's file-naming convention
- benchmark, as it will introduce some penalty

Reply:
I did few benchmarks:
`[info] Benchmark                    (count)   Mode  Cnt   Score   Error  Units
[info] AppendBenchmarks.zio             500  thrpt   30   9.986 ± 0.026  ops/s ## my branch
[info] AppendBenchmarks.zio             500  thrpt   30  10.071 ± 0.035  ops/s

[info] Benchmark          (count)   Mode  Cnt   Score   Error  Units
[info] SetBenchmarks.zio      500  thrpt   30   9.918 ± 0.025  ops/s           ## my branch
[info] SetBenchmarks.zio      500  thrpt   30  10.426 ± 0.037  ops/s

[info] Benchmark          (count)   Mode  Cnt  Score   Error  Units
[info] GetBenchmarks.zio      500  thrpt   30  9.358 ± 0.631  ops/s            ## my branch
[info] GetBenchmarks.zio      500  thrpt   30  9.927 ± 0.023  ops/s

[info] Benchmark           (count)   Mode  Cnt   Score   Error  Units
[info] IncrBenchmarks.zio      500  thrpt   30   9.931 ± 0.203  ops/s          ## my branch
[info] IncrBenchmarks.zio      500  thrpt   30  10.188 ± 0.203  ops/s`

Is it enough?

Reply:
>submit it in a separate pull request as it is not related to slots and keys

It's related to slot and keys, we have to differentiate parameter types in executor

Reply:
>either split it to multiple files or respect scala's file-naming convention

Splitted

Reply:
> >submit it in a separate pull request as it is not related to slots and keys
> 
> It's related to slot and keys, we have to differentiate parameter types in executor

It causes a ripple effect of changes throughout the entire library. I would argue that this is not a "local" change :)

Reply:
I agree with you it is not a "local" change - there is no argues.) I don't know how to solve that issue do not change inner architecture 😇 
But current architecture do not have tools to differentiate Redis (resp) parameter types, that is necessary for advanced client library. If we want to have flexible, easy maintainable executors we need it. Something similar is implemented in the lettuce, i get basic idea from it.

Reply:
I am with you when it comes to the approach to solve it, just do the RespCommand part in one PR and "apply" it another :)

Reply:
Do you want that?) 
https://github.com/zio/zio-redis/pull/763